projects/fsastorefrontlib/src/cms-components/my-account/inbox/inbox-tab/inbox-messages/inbox-messages.component.ts

```diff
@@ -60,6 +61,9 @@ export class InboxMessagesComponent implements OnInit, OnDestroy {
               group.messageGroup !== this.messageGroup
             ) {
               this.clearSearchData();
+              this.envelopState = false;
+              this.mainCheckboxChecked = false;
```

Comment:
I think it would be better to reset mainCheckboxChecked variable after switching on the new tab. Otherwise we have a bug.